### PR TITLE
Add WrapError to automatically wrap error and log

### PR DESCRIPTION
**Pull request by sunfmin**
How about this?

https://github.com/theplant/appkit/issues/4

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -25,6 +25,16 @@ func (l Logger) With(keysvals ...interface{}) Logger {
 }
 
 /*
+WrapError can wrap error with kerrs to structured log
+*/
+func (l Logger) WrapError(err error) log.Logger {
+	if err == nil {
+		return l
+	}
+	return l.WithError(kerrs.Wrapv(err, ""))
```
Comment (bodhi) [inline log/log.go]: Why do you need to pass an empty string here?

File: log/log_test.go [hunk 2]
```diff
@@ -90,6 +90,8 @@ func TestHuman(t *testing.T) {
 		t.Error(err)
 	}
 
+	l.WrapError(errors.New("hello error")).Log("msg", "there is a big error")
+
```
Comment (bodhi) [inline log/log_test.go]: What does `TestHuman` actually test?

File: log/log.go [hunk 3]
```diff
@@ -25,6 +25,16 @@ func (l Logger) With(keysvals ...interface{}) Logger {
 }
 
 /*
+WrapError can wrap error with kerrs to structured log
+*/
+func (l Logger) WrapError(err error) log.Logger {
```
Comment (bodhi) [inline log/log.go]: `WrapError` and `WithError` have the same signature and basically the same description. How would I choose which one to use?

File: log/log_test.go (hunk 2 above)
Comment (sunfmin) [reply log/log_test.go]: test grammer~ and give you a easy way to see what output looks like...

File: log/log.go (hunk 3 above)
Comment (sunfmin) [reply log/log.go]: Any suggestions to improve the name? I think when you see `Wrap` you know it wraps.. then you know that `WithError` don't wrap.

File: log/log.go (hunk 1 above)
Comment (sunfmin) [reply log/log.go]: Wrapv takes a message parameter. what should we pass?

**Comment by sunfmin** [comment]
Add better comments, Anything else to change? @bodhi

**Pull request merged**